lib/parser.rb

```diff
@@ -13,18 +26,46 @@ def first_line
   end
   
   def generate_json
-    file = @path.split("/").last
+  file = @path.split("/").last
+  
+  obj = { file => {
+    :lines => count_lines(),
+    :players => count_players()
+    }}
     
-    obj = { file => {
-      :lines => count_lines()
-      } }
-      
-    JSON.pretty_generate(obj)
+  JSON.pretty_generate(obj)
+  end
+
+  private
+
+  def count_lines
+    data = File.readlines(@path).count
+  end  
+
+  def count_players
+    players = []
+    File.readlines(@path).each do |line|
+      PATTERNS.each do |pattern|
+        temp = line_scanner(line, pattern[:word], pattern[:init], pattern[:end])
+
+        player_include(temp, players)
+      end
     end
 
-    private
+    players
+  end
+
+  def line_scanner(line, word, search_init, search_end)
+    if line.include?(word)
+      line.split(search_init).last.split(search_end).first
+    end
+  end
 
-    def count_lines
-      data = File.readlines(@path).count
+  def player_include(temp, players)
+    unless temp.nil?
```

Comment:
here you can use [guard cluase](https://anthonygharvey.com/guard-clauses-vs-nested-conditionals) to avoid this unless under unless block